embassy-stm32l0/src/exti.rs

```diff
@@ -0,0 +1,232 @@
+use core::future::Future;
+use core::mem;
+use core::pin::Pin;
+
+use embassy::interrupt::Interrupt;
+use embassy::traits::gpio::{WaitForAnyEdge, WaitForFallingEdge, WaitForRisingEdge};
+use embassy::util::InterruptFuture;
+
+use crate::hal::{
+    exti::{Exti, ExtiLine, GpioLine, TriggerEdge},
+    gpio,
+    syscfg::SYSCFG,
+};
+use crate::interrupt;
+use crate::pac::EXTI;
+
+pub struct ExtiManager {
+    syscfg: SYSCFG,
+}
+
+impl<'a> ExtiManager {
+    pub fn new(_exti: Exti, syscfg: SYSCFG) -> Self {
+        Self { syscfg }
+    }
+
+    pub fn new_pin<T, I>(&'static mut self, pin: T, interrupt: I) -> ExtiPin<T, I>
+    where
+        T: PinWithInterrupt<Interrupt = I>,
+        I: Interrupt,
+    {
+        ExtiPin {
+            pin,
+            interrupt,
+            mgr: self,
+        }
+    }
+}
+
+pub struct ExtiPin<T, I> {
```

Comment:
Remove the `I` type parameter by adding type bounds to the `Interrupt` type. See stm32f4 for an example.